extension/src/test/util/experiments.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  Executor,
+  ExecutorState,
+  ExpData,
+  EXPERIMENT_WORKSPACE_ID,
+  ExperimentStatus,
+  ExpRange,
+  ExpShowOutput,
+  ExpState
+} from '../../cli/dvc/contract'
+
+export const generateTestExpData = (data: Partial<ExpData> = {}): ExpData => ({
+  deps: null,
+  meta: { has_checkpoints: false },
+  metrics: null,
+  outs: null,
+  params: null,
+  rev: EXPERIMENT_WORKSPACE_ID,
+  timestamp: null,
+  ...data
+})
+
+const generateTestExpState = (
+  rev: string,
+  data: Partial<ExpData> = {},
+  name?: string
+): ExpState => ({
+  data: generateTestExpData({ ...data, rev }),
+  name,
+  rev
+})
+
+const isTestExpWithName = (
+  testExp:
+    | { rev: string; name: string; data: Partial<ExpData> }
+    | Partial<ExpData>
+): testExp is { rev: string; name: string; data: Partial<ExpData> } =>
+  !!(testExp as { name: string }).name
+
+const getExecutorState = (executor: Partial<ExecutorState>): ExecutorState => {
+  if (!executor) {
+    return null
+  }
+  return {
+    local: null,
+    name: Executor.WORKSPACE,
+    state: ExperimentStatus.RUNNING,
+    ...executor
+  }
+}
+
+const generateBaselineWithExperiments = ({
+  data = {},
+  name,
+  rev,
+  testExps
+}: {
+  data?: Partial<ExpData>
+  name?: string
+  rev: string
+  testExps: (
+    | {
+        data: Partial<ExpData>
+        executor: Partial<ExecutorState>
+        name: string
+        rev: string
+      }
```

Comment:
I'm seeing multiple ` { /* obj with various keys */ } | Partial<ExpData>` within these utils. Should these "obj with various keys" be moved into their own types? Might make things more readable 🤔

Reply:
I'll follow up on these comments